app/screens/select_server/index.js

```diff
@@ -5,25 +5,24 @@ import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
 import {getPing, resetPing} from 'mattermost-redux/actions/general';
-import {RequestStatus} from 'mattermost-redux/constants';
 
 import {setLastUpgradeCheck} from 'app/actions/views/client_upgrade';
+import {loadConfigAndLicense} from 'app/actions/views/root';
 import {handleServerUrlChanged} from 'app/actions/views/select_server';
 import getClientUpgrade from 'app/selectors/client_upgrade';
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
 import SelectServer from './select_server';
 
 function mapStateToProps(state) {
-    const {config: configRequest, license: licenseRequest} = state.requests.general;
     const {config, license} = state.entities.general;
     const {currentVersion, latestVersion, minVersion} = getClientUpgrade(state);
 
     return {
         ...state.views.selectServer,
         config,
         currentVersion,
-        hasConfigAndLicense: configRequest.status === RequestStatus.SUCCESS && licenseRequest.status === RequestStatus.SUCCESS,
+        hasConfigAndLicense: Object.keys(config).length > 0 && Object.keys(license).length > 0,
```

Comment:
Does this get properly reset if you select a server, go back to the login page, and then enter the URL of a different server?

Reply:
very good point